components/home/reducers.js

```diff
@@ -1,34 +1,41 @@
 import { UPDATE_REACTION, REQUEST_UPLOAD_LIST } from './actionTypes';
-import { INITIAL_STATE, USER_UPLOADS } from './constants';
+import { INITIAL_STATE } from './constants';
 
-/**
- * Creates a Javascript Map with the user uploads mapped by id
- *
- * @param {Array} USER_UPLOADS - a users uploads
- * @return {Map} - the user uploads
- */
+import fakeUploads from '../../models/fakeUploads';
 
+/**
+* Creates a Javascript Map with the user uploads mapped by id
+*
+* @param {Array} fakeUploads - a users uploads
+* @return {Map} - the user uploads
+*/
 function generateUploadsMap() {
     const uploads = new Map();
 
-    USER_UPLOADS.forEach(userUpload => {
-        const { id } = userUpload;
-
-        uploads.set(id, userUpload);
+    fakeUploads.forEach(upload => {
+        const { id } = upload;
+        uploads.set(id, upload);
     });
 
     return uploads;
 }
 
 function updateUploadReaction(id, type, uploads) {
     const updatedUploads = new Map([...uploads.entries()]);
-    const userUpload = updatedUploads.get(id);
-    if (!userUpload.reactions[type]) {
-        userUpload.reactions[type] += 1;
-    } else {
-        userUpload.reactions[type] -= 1;
-    }
-    updatedUploads.set(id, userUpload);
+    const upload = updatedUploads.get(id);
+
+    uploads.forEach(() => {
+        const { users: { userId } } = upload.reactions[type];
+        if (!userId) {
```

Comment:
this doesn't make any sense Chioma, why are you checking for the `userId` key in users?

Reply:
The object is supposed to save user Ids that have reacted to a type of reaction, so the idea is to check if a particular user Id exists in that object.

Reply:
Are you checking for a particular user id or a key called `userId`? Those are not the same thing